homeassistant/components/ezviz/__init__.py

```diff
@@ -50,69 +55,71 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-            # Should only execute on addition of new camera entry.
-            # Fetch Entry id of main account and reload it.
-            for item in hass.config_entries.async_entries():
-                if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
-                    _LOGGER.info("Reload EZVIZ integration with new camera rtsp entry")
-                    await hass.config_entries.async_reload(item.entry_id)
+    # Initialize EZVIZ cloud entities
+    if PLATFORMS_BY_TYPE[sensor_type]:
+        # Initiate reauth config flow if account token if not present.
+        if not entry.data.get(CONF_SESSION_ID):
+            raise ConfigEntryAuthFailed
+
+        ezviz_client = EzvizClient(
+            token={
+                CONF_SESSION_ID: entry.data.get(CONF_SESSION_ID),
+                CONF_RFSESSION_ID: entry.data.get(CONF_RFSESSION_ID),
+                "api_url": entry.data.get(CONF_URL),
+            },
+            timeout=entry.options.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        )
+
+        try:
+            await hass.async_add_executor_job(ezviz_client.login)
 
-        return True
+        except (EzvizAuthTokenExpired, EzvizAuthVerificationCode) as error:
+            raise ConfigEntryAuthFailed from error
 
-    try:
-        ezviz_client = await hass.async_add_executor_job(
-            _get_ezviz_client_instance, entry
+        except (InvalidURL, HTTPError, PyEzvizError) as error:
+            _LOGGER.error("Unable to connect to Ezviz service: %s", str(error))
+            raise ConfigEntryNotReady from error
+
+        coordinator = EzvizDataUpdateCoordinator(
+            hass, api=ezviz_client, api_timeout=entry.options[CONF_TIMEOUT]
         )
-    except (InvalidURL, HTTPError, PyEzvizError) as error:
-        _LOGGER.error("Unable to connect to EZVIZ service: %s", str(error))
-        raise ConfigEntryNotReady from error
 
-    coordinator = EzvizDataUpdateCoordinator(
-        hass, api=ezviz_client, api_timeout=entry.options[CONF_TIMEOUT]
-    )
-    await coordinator.async_refresh()
+        await coordinator.async_config_entry_first_refresh()
+
+        hass.data[DOMAIN][entry.entry_id] = {DATA_COORDINATOR: coordinator}
 
-    if not coordinator.last_update_success:
-        raise ConfigEntryNotReady
+    entry.async_on_unload(entry.add_update_listener(_async_update_listener))
 
-    undo_listener = entry.add_update_listener(_async_update_listener)
+    # Check EZVIZ cloud account entity is present, reload cloud account entities for camera entity change to take effect.
+    # Cameras are accessed via local RTSP stream with unique credentials per camera.
+    # Separate camera entities allow for credential changes per camera.
+    if sensor_type == ATTR_TYPE_CAMERA and hass.data[DOMAIN]:
+        for item in hass.config_entries.async_entries(domain=DOMAIN):
+            if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
+                _LOGGER.info("Reload Ezviz main account with camera entry")
```

Comment:
Debug level please.

Reply:
Addressed in https://github.com/home-assistant/core/pull/94384